app/src/main/java/org/openobservatory/ooniprobe/common/ProxySettings.java

```diff
@@ -72,21 +72,22 @@ private boolean isIPv6(String hostname) {
 
     /** getProxyString returns to you the proxy string you should pass to oonimkall. */
     public String getProxyString() throws URISyntaxException {
-        if (protocol == ProxyProtocol.NONE)
+        if (protocol == ProxyProtocol.NONE) {
             return "";
-        if (protocol == ProxyProtocol.PSIPHON)
+        } else if (protocol == ProxyProtocol.PSIPHON) {
             return "psiphon://";
-        if (protocol == ProxyProtocol.SOCKS5) {
+        } else if (protocol == ProxyProtocol.SOCKS5||protocol == ProxyProtocol.HTTP||protocol == ProxyProtocol.HTTPS) {
             // Alright, we now need to construct a new SOCKS5 URL. We are going to defer
             // doing that to the Java standard library (er, the Android stdlib).
-            String urlStr = "socks5://" + hostname + ":" + port + "/";
+            String urlStr = protocol+"://" + hostname + ":" + port + "/";
             if (isIPv6(hostname)) {
-                urlStr = "socks5://[" + hostname + "]:" + port + "/"; // IPv6 must be quoted in URLs
+                urlStr = protocol+"://[" + hostname + "]:" + port + "/"; // IPv6 must be quoted in URLs
             }
             URI url = new URI(urlStr);
             return url.toASCIIString();
+        } else {
+            return "";
```

Comment:
You don't need this else because the previous `if` branch is already return terminated